src/models/ratings/ratings.service.ts

```diff
@@ -0,0 +1,434 @@
+import { HttpException, Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Like, MoreThan, Repository } from 'typeorm';
+import { Ratings } from './entities/ratings.entity';
+import { RatingQuery } from './dto/ratings.dto';
+import { access, readFile, writeFile } from 'fs/promises';
+import { Players } from './entities/players.entity';
+import { Games } from '../games-history/entities/games.entity';
+
+@Injectable()
+export class RatingService {
+	private readonly logger = new Logger(RatingService.name);
+
+	constructor(
+		@InjectRepository(Ratings, 'default')
+		private ratingRespository: Repository<Ratings>,
+		@InjectRepository(Players, 'default')
+		private playersRepository: Repository<Players>,
+		@InjectRepository(Games, 'games')
+		private gamesRepository: Repository<Games>,
+	) {}
+
+	async getAll(query?: RatingQuery): Promise<any> {
+		const limit = parseInt(query.limit) || 50;
+		const skip = parseInt(query.skip) || 0;
+		const page = parseInt(query.page) || 0;
+		const order: 'ASC' | 'DESC' = query.order || 'DESC';
+		const sort = query.sort ? query.sort : 'rating';
+		const whereSearch = {
+			ratingbase: 0,
+			unrated: 0,
+			ratedgames: MoreThan(0),
+		}
+		if (query.name) whereSearch['name'] = Like(`${query.name}`);
+		try {
+			const results = await this.ratingRespository.findAndCount({
+				select: ['name', 'rating', 'ratedgames', 'maxrating', 'isbot'],
+				where: whereSearch,
+				order: {
+					[sort]: order,
+				},
+				take: limit,
+				skip: limit * page || skip,
+			});
+			return {
+				items: results[0] || [],
+				total: results[1] || 0,
+				page: page + 1,
+				perPage: limit,
+				totalPages: Math.ceil(results[1] / limit),
+			};
+		} catch (error) {
+			console.error(error);
```

Comment:
why not use `this.logger.error`?

Reply:
Just forgot to move it over to logger, which I don't using but it does strip out the line number where the error happened which makes it harder to debug...

Reply:
I think the `console.log` can go the next time you need to touch this file 😄